robot/Cumulus/resources/OpportunityPageObject.py

```diff
@@ -31,6 +32,19 @@ def navigate_to_writeoff_payments_page(self):
         self.npsp.wait_for_locator('frame','Write Off Remaining Balance')
         self.npsp.choose_frame("Write Off Remaining Balance")
         self.selenium.wait_until_page_contains("You are preparing to write off")
+        
+    def change_related_contact_role_settings(self,name,role=None,**kwargs):
+        
+        dropdown = npsp_lex_locators['related_drop_down'].format(name)
+        edit = npsp_lex_locators['record']['dd_edit_option'].format("Edit")
+        self.selenium.wait_until_page_contains_element(dropdown)
+        self.salesforce._jsclick(dropdown)
+        self.selenium.wait_until_element_is_visible(edit)
+        self.selenium.click_element(edit)
+        time.sleep(2)
```

Comment:
can we check on something instead of adding sleep?

Reply:
Agree and changed to wait_for_modal